### PR TITLE
[Document][991947] Add flag to document controller to allow the prefe…

### PR DESCRIPTION
**Pull request by viniciusjarina**
Add flag to document controller to allow the preferred the external binding as preferred viewer.

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentControllerDescription.cs [hunk 1]
```diff
@@ -40,6 +40,12 @@ public class DocumentControllerDescription
 		/// </summary>
 		public bool CanUseAsDefault { get; set; }
 
+		/// <summary>
+		/// Change the order of document controllers and external binding if any of the document controllers flag
+		/// it prefer that document prefer to be opened in the external binding.
+		/// </summary>
+		public bool PreferExtenalBinding { get; set; }
+
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentControllerDescription.cs]: Typo. PreferExternalBinding.

**Comment by viniciusjarina** [comment]
So as discussed with @slluis. For release 8.4 we are landingn the `PreferExternalBinding` flag, but for the next release we will refactor the API to provide a `Priority` value.

**Comment by viniciusjarina** [comment]
@monojenkins backport release-8.4

**Comment by slluis** [comment]
@monojenkins backport release-8.4

**Pull request merged**